server/src/main/java/org/elasticsearch/cluster/routing/RecoverySource.java

```diff
@@ -252,6 +278,9 @@ protected void writeAdditionalFields(StreamOutput out) throws IOException {
             snapshot.writeTo(out);
             Version.writeVersion(version, out);
             out.writeString(index);
```

Comment:
can we avoid sending this field if `indexId != null` (i.e. in the non-bwc case)

Reply:
Not sure, see the other question, seems to me we need to write the uuid string as optional, maybe not?